titus-supplementary-component/task-relocation/src/main/java/com/netflix/titus/supplementary/relocation/connector/NodePredicates.java

```diff
@@ -64,11 +65,39 @@ static boolean hasBadCondition(V1Node node, Function<String, Matcher> badConditi
         return false;
     }
 
+    @VisibleForTesting
+    static boolean hasBadTaint(V1Node node, Function<String, Matcher> badTaintExpression,
+                                   int nodeTaintTransitionTimeThresholdSeconds) {
+        if (node.getSpec() != null && node.getSpec().getTaints() != null) {
+            return node.getSpec().getTaints().stream()
+                    .anyMatch(v1Taint -> badTaintExpression.apply(v1Taint.getKey()).matches() &&
+                            matchesTaintValueIfAvailable(v1Taint, Boolean.toString(true)) &&
+                            !isTransitionedRecently(v1Taint.getTimeAdded(), nodeTaintTransitionTimeThresholdSeconds));
+        }
+        return false;
+    }
+
+    static boolean matchesTaintValueIfAvailable(V1Taint taint, String value) {
+        if (taint.getValue() != null) {
+            return taint.getValue().equalsIgnoreCase(value);
+        }
+        return true;
+    }
+
+
     static boolean isNodeConditionTransitionedRecently(V1NodeCondition nodeCondition, int thresholdSeconds) {
         DateTime threshold = DateTime.now().minusSeconds(thresholdSeconds);
```

Comment:
by using the Titus commons Clock here, you will have more flexibility over injecting in and manipulating the time.